lib/bolt/bolt_option_parser.rb

```diff
@@ -298,15 +296,24 @@ def initialize(options)
       update
     end
 
-    def hide_target_opts(toggle = true)
-      @nodes.hide = @query.hide = @rerun.hide = @targets.hide = toggle
+    def remove_excluded_opts
+      # Remove any options that are not available for the specified subcommand
+      top.list.delete_if do |opt|
+        opt.respond_to?(:switch_name) && EXCLUDES.key?(@options[:subcommand]) &&
+          EXCLUDES[@options[:subcommand]].include?(opt.switch_name)
+      end
+      # Remove any separators if all options of that type have been removed
```

Comment:
I think you could avoid iterating over all options twice (well I guess 4 times because currently  `remove_exclude_opts` is called twice see above comment) with something like
```ruby
      top.list.reject!.with_index do |opt, index| 
        (opt.respond_to?(:switch_name) && EXCLUDES.key?(@options[:subcommand]) &&
        EXCLUDES[@options[:subcommand]].include?(opt.switch_name)) || 
        (opt.is_a?(String) && top.list[index + 1].is_a?(String))
      end
```
Obviously that is super messy and we cram quite a bit of logic in there, but you could explore something like that.

Reply:
I tried something like that previously but it leaves behind separators unfortunately. Since the switches are left in place until the iteration is complete, there's never an instance where the separators are next to each other in `top.list`. If another method is used that removes switches as we go, then that causes some switches to be skipped since they're shifted during the iteration.

Reply:
Oh, yeah I guess modifying lists as you are iterating on them is bad anyway. That was horrible advice. My bad.